onnxruntime/core/optimizer/optimizer_execution_frame.cc

```diff
@@ -145,15 +145,15 @@ Status OptimizerExecutionFrame::CreateNodeOutputMLValueImpl(OrtValue& ort_value,
   }
 
   if (ml_type->IsTensorSequenceType()) {
-    auto element_type = ml_type->AsSequenceTensorBase()->GetElementType();
+    auto element_type = ml_type->AsSequenceTensorType()->GetElementType();
     auto p_sequence = std::make_unique<TensorSeq>(element_type);
     auto ml_tensor_sequence = DataTypeImpl::GetType<TensorSeq>();
     ort_value.Init(p_sequence.release(), ml_tensor_sequence, ml_tensor_sequence->GetDeleteFunc());
     return Status::OK();
   }
 
   if (!ml_type->IsTensorType()) {
-    assert(ml_type->AsNonTensorTypeBase() != nullptr);
+    assert(ml_type->AsNonTensorType() != nullptr);
```

Comment:
>_type->AsNonTensorType() [](http://example.com/codeflow?start=13&length=24)

I think we better do just about, use AsNonTensorBase() instead of the cast.